containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -243,16 +243,19 @@ class UnorderedMap {
   using const_map_type = UnorderedMap<const_key_type, const_value_type,
                                       device_type, hasher_type, equal_to_type>;
 
-  static const bool is_set = std::is_void<value_type>::value;
-  static const bool has_const_key =
-      std::is_same<const_key_type, declared_key_type>::value;
-  static const bool has_const_value =
-      is_set || std::is_same<const_value_type, declared_value_type>::value;
+  static constexpr bool is_set = std::is_void_v<value_type>;
+  static constexpr bool has_const_key =
+      std::is_same_v<const_key_type, declared_key_type>;
+  static constexpr bool has_const_value =
+      is_set || std::is_same_v<const_value_type, declared_value_type>;
 
-  static const bool is_insertable_map =
+  static constexpr bool is_insertable_map =
       !has_const_key && (is_set || !has_const_value);
-  static const bool is_modifiable_map = has_const_key && !has_const_value;
-  static const bool is_const_map      = has_const_key && has_const_value;
+  static constexpr bool is_modifiable_map = has_const_key && !has_const_value;
+  static constexpr bool is_const_map      = has_const_key && has_const_value;
+
+  static constexpr size_type invalid_index =
+      KOKKOS_INVALID_INDEX_TYPE(size_type);
```

Comment:
Why is it important for `invalid_index` to be public? How would it be used in user code?

Reply:
In one of my tests, I do:
```c++
using map_t = Kokkos::UnorderedMap<int, double, execution_space>;

...

map_t map(...);

...

Kokkos::parallel_for(
    ...,
    KOKKOS_LAMBDA(...) {

        ....

        if(const auto direct_index = map.find(my_key); direct_index != ~static_cast<typename umap_t::size_type>(0))
        {
            <use `direct_index` here>
        }
    }
);

...
```

So in this case it would be nice to have access to `invalid_index`, which in fact is a value that can be returned by `find`, which is **a public member function**. IMO we don't even need a use case to decide if `invalid_index` should be public. I think it should because it can be returned by a public member function.